runtime-sdk/src/modules/core/types.rs

```diff
@@ -32,3 +32,37 @@ pub struct CallDataPublicKeyQueryResponse {
     /// Public key used for deriving the shared secret for encrypting call data.
     pub public_key: SignedPublicKey,
 }
+
+#[derive(Debug, Copy, Clone, cbor::Encode, cbor::Decode)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+pub enum MethodHandlerKind {
+    Call,
+    // `Prefetch` is omitted because it is an implementation detail of handling `Call`s.
+    Query,
+    MessageResult,
```

Comment:
Maybe it would make sense to rename the serialized representation of these variants into lower case, e.g.:
* `call`
* `query`
* `message-result`